transcendence-app/src/chat/chatroom/infrastructure/db/postgres/chatroom.postgres.repository.ts

```diff
@@ -103,6 +138,104 @@ export class ChatroomPostgresRepository
+                       ORDER BY cm.${chatMessageKeys.CREATED_AT}),
+               lChatMessages as (SELECT u.${userKeys.AVATAR_ID},
+                                        u.${userKeys.AVATAR_X},
+                                        u.${userKeys.AVATAR_Y},
+                                        'chat/' || u.${userKeys.USERNAME}          AS "url",
+                                        u.${userKeys.USERNAME}                     AS "name",
+                                        md."lastMsgSenderUsername",
+                                        md.${chatMessageKeys.CONTENT}::varchar(20) AS "lastMessage",
+                                        md.${chatMessageKeys.CREATED_AT}           AS "lastMessageDate"
+                                 FROM ${table.USERS} u
+                                          INNER JOIN msgData md ON u.${userKeys.ID} = md."userId"
+                                 ORDER BY md.${chatMessageKeys.CREATED_AT}),
+               lchatrooms as
+                   (SELECT c.*
+                    FROM ${this.table} c
+                             INNER JOIN ${table.CHATROOM_MEMBERS} cm
+                                        ON cm."userId" = $1
+                                            AND cm."chatId" = c."id"
+                    WHERE cm."joinedAt" IS NOT NULL),
+               crMsg AS (SELECT m."chatroomId",
+                                m."userId",
+                                m."id",
+                                m."createdAt"
+                         FROM ${table.CHATROOM_MESSAGE} m
+                                  INNER JOIN "lchatrooms" cr ON cr."id" = m."chatroomId"),
+               crDateProvider AS (SELECT m."chatroomId",
+                                         ROW_NUMBER() OVER (
+                                             PARTITION BY m."chatroomId" ORDER BY m."createdAt" DESC
+                                             ) AS "rowNumber",
+                                         m."id"
+                                  FROM crmsg m),
+               crMsgIds AS (SELECT dp."id" AS "msgId"
+                            FROM crDateProvider dp
+                            WHERE dp."rowNumber" = 1),
+               crMsgData AS (SELECT cm."chatroomId",
+                                    u."username" AS "lastMsgSenderUsername",
+                                    cm."content",
+                                    cm."createdAt"
+                             FROM ${table.CHATROOM_MESSAGE} cm
+                                      INNER JOIN crmsgIds mi ON cm."id" = mi."msgId"
+                                      LEFT JOIN ${table.USERS} u ON cm."userId" = u."id"
+                             ORDER BY cm."createdAt"),
+               lChatroomMessages AS (SELECT cr."avatarId",
+                                            cr."avatarX",
+                                            cr."avatarY",
+                                            'chatroom/' || cr."id"    AS "url",
```

Comment:
Again:
maybe we could pass this as a parameter in the incoming request body so that webapp can own what the route of its own pages are.